pkg/gather/clusterconfig/csr.go

```diff
@@ -0,0 +1,264 @@
+package clusterconfig
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"github.com/openshift/insights-operator/pkg/utils"
+	"k8s.io/api/certificates/v1beta1"
+	certificatesv1b1api "k8s.io/api/certificates/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/klog"
+)
+
+type CSRAnonymizer struct {
+	*CSRAnonymizedFeatures
+}
+
+func (a CSRAnonymizer) Marshal(_ context.Context) ([]byte, error) {
+	// json.Marshal can handle nil well
+	return json.Marshal(a.CSRAnonymizedFeatures)
+}
+
+type CSRs struct {
+	Requests   []v1beta1.CertificateSigningRequest
+	Anonymized []CSRAnonymizer
+}
+
+func FromCSRs(requests *v1beta1.CertificateSigningRequestList) *CSRs {
+	return &CSRs{Requests: requests.Items}
+}
+
+func (c *CSRs) Anonymize() *CSRs {
+	res := &CSRs{}
+	for _, r := range c.Requests {
+		af := anonymizeCSR(&r)
+		res.Anonymized = append(res.Anonymized, CSRAnonymizer{af})
+	}
+	return res
+}
+
+func (c *CSRs) Filter(f FilterFeatures) *CSRs {
+	res := &CSRs{}
+	for _, r := range c.Anonymized {
+		if f(r.CSRAnonymizedFeatures) {
+			res.Anonymized = append(res.Anonymized, r)
+		}
+	}
+	return res
+}
+
+func (c *CSRs) Select() ([]CSRAnonymizer, error) {
+	return c.Anonymized, nil
+}
+
+type FilterFeatures func(c *CSRAnonymizedFeatures, opt ...FilterOptFunc) bool
+
+type FilterOpt struct {
+	time time.Time
+}
+
+type FilterOptFunc = func(o *FilterOpt)
+
+func WithTime(t time.Time) FilterOptFunc {
+	return func(o *FilterOpt) {
+		o.time = t
+	}
+}
+
+func IncludeCSR(c *CSRAnonymizedFeatures, opts ...FilterOptFunc) bool {
+	opt := &FilterOpt{time: time.Now()}
+	for _, o := range opts {
+		o(opt)
```

Comment:
these variables names are not good, e, t, o, i, etc, it could be a little more descriptive. In this case it could be `optFn`.